internal/handlers/scan_sbom.go

```diff
@@ -119,8 +134,47 @@ func (h *ScanSBOMHandler) Handle(ctx context.Context, message []byte) error { //
 		"--db-repository", "public.ecr.aws/aquasecurity/trivy-db",
 		"--java-db-repository", "public.ecr.aws/aquasecurity/trivy-java-db",
 		"--output", reportFile.Name(),
-		sbomFile.Name(),
-	})
+	}
+	// Set XDG_DATA_HOME environment variable to /tmp because trivy expects
+	// the repository file in that location and there is no way to change it
+	// through input flags:
+	// https://trivy.dev/v0.64/docs/supply-chain/vex/repo/#default-configuration
+	// TODO(alegrey91): fix upstream
+	trivyHome, err := os.MkdirTemp("/tmp", "trivy-")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary trivy home: %w", err)
+	}
+	err = os.Setenv("XDG_DATA_HOME", trivyHome)
+	if err != nil {
+		return fmt.Errorf("failed to set XDG_DATA_HOME to %s: %w", trivyHome, err)
+	}
+
+	if len(vexHubList.Items) > 0 {
+		trivyVEXPath := path.Join(trivyHome, trivyVEXSubPath)
+		vexRepoPath := path.Join(trivyVEXPath, trivyVEXRepoFile)
+		if err = h.setupVEXHubRepositories(vexHubList, trivyVEXPath, vexRepoPath); err != nil {
+			return fmt.Errorf("failed to setup VEX Hub repositories: %w", err)
+		}
+		// Clean up the repository.yaml file after each handler execution to
+		// ensure VEX repositories are refreshed on every run.
```

Comment:
This comment is outdated with the latest changes. Here, we are cleaning up the entire temp directory used as the Trivy home.